src/global.h

```diff
@@ -362,15 +362,214 @@ class CCustomEvent : public QEvent
+//============================================================================
+#ifndef HEADLESS
+#    include <QMessageBox>
+#else
+#    define QDialog void
+#endif
+
+// html text macro's (for use in gui texts)
+#define htmlBold( T ) "<b>" + T + "</b>"
+#define htmlNewLine() "<br>"
+
+class CMsgBoxes
+{
+protected:
+    static QDialog* pMainForm;
+    static QString  strMainFormName;
+
+public:
+    static void init ( QDialog* theMainForm, QString theMainFormName )
+    {
+        pMainForm       = theMainForm;
+        strMainFormName = theMainFormName;
+    }
+
+    static QDialog*       MainForm() { return pMainForm; }
+    static const QString& MainFormName() { return strMainFormName; }
+
+    // Message boxes:
+    static void ShowError ( QString strError );
+    static void ShowWarning ( QString strWarning );
+    static void ShowInfo ( QString strInfo );
+};
+
+//============================================================================
+// CCommandlineOptions class:
+//  Note that passing commandline arguments to classes is no longer required,
+//  since via this class we can get commandline options anywhere.
+//============================================================================
+
+class CCommandlineOptions
+{
+public:
+    CCommandlineOptions() { reset(); }
+
+private:
+    friend int main ( int argc, char** argv );
+
+    // Statics assigned from main ()
+    static int    appArgc;
+    static char** appArgv;
+
+public:
+    static QString GetProgramPath() { return QString ( *appArgv ); }
+
+public:
+    // sequencial parse functions using the argument index:
+
+    static bool GetFlagArgument ( int& i, const QString& strShortOpt, const QString& strLongOpt );
+
+    static bool GetStringArgument ( int& i, const QString& strShortOpt, const QString& strLongOpt, QString& strArg );
+
+    static bool GetNumericArgument ( int&           i,
+                                     const QString& strShortOpt,
+                                     const QString& strLongOpt,
+                                     double         rRangeStart,
+                                     double         rRangeStop,
+                                     double&        rValue );
+
+public:
+    // find and get a specific argument:
+
+    static bool GetFlagArgument ( const QString& strShortOpt, const QString& strLongOpt )
+    {
+        int i = 1;
+        while ( i < appArgc )
+        {
+            if ( GetFlagArgument ( i, strShortOpt, strLongOpt ) )
+            {
+                return true;
+            }
+
+            i++;
+        }
+
+        return false;
+    }
+
+    static bool GetStringArgument ( const QString& strShortOpt, const QString& strLongOpt, QString& strArg )
+    {
+        int i = 1;
+        while ( i < appArgc )
+        {
+            if ( GetStringArgument ( i, strShortOpt, strLongOpt, strArg ) )
+            {
+                return true;
+            }
+
+            i++;
+        }
+
+        return false;
```

Comment:
Looks a bit like we could use a for loop?

Reply:
> Looks a bit like we could use a for loop?

Yes could just as well be a for loop...